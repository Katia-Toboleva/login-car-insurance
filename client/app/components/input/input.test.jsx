import React from 'react';
import { mount } from 'enzyme';
import Input from './input';

const getUnit = (props) => (
  mount(<Input {...props} />)
);

describe('Input', () => {
  describe('render', () => {
    it('should render correctly', () => {
      const wrapper = getUnit();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('handleChange', () => {
    it('updates html correctly', () => {
      const wrapper = getUnit();
      const value = 'a';
      wrapper.find('.input__body').simulate('change', {
        target: { value },
      });

      expect(wrapper).toMatchSnapshot();
    });

    it('calls props.onChange with correct value', () => {
      const cb = jest.fn();
      const value = 'a';
      const wrapper = getUnit({ onChange: cb, value });
      wrapper.find('input').simulate('change', {
        target: { value },
      });

      expect(cb).toHaveBeenCalledTimes(1);
      expect(cb).toHaveBeenCalledWith(value);
    });
  });

  describe('handleInputFocus', () => {
    it('updates html correctly', () => {
      const wrapper = getUnit();
      wrapper.find('.input__body').simulate('focus');

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('handleInputBlur', () => {
    it('updates html correctly', () => {
      const wrapper = getUnit();
      wrapper.find('.input__body').simulate('blur');

      expect(wrapper).toMatchSnapshot();
    });
  });
});
