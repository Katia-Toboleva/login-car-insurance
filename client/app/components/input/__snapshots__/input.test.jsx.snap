// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input handleChange updates html correctly 1`] = `
<Input
  onChange={[Function]}
  placeholder="text"
  value=""
>
  <div
    className="input input--focused"
  >
    <span
      className="input__header"
    >
      <Text
        center={false}
        color="black"
        size="medium"
        text="text"
        transform="capitalize"
      >
        <div
          className="text text--size-medium text--color-black text--transform-capitalize"
        >
          <span>
            text
          </span>
        </div>
      </Text>
    </span>
    <input
      className="input__body"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="text"
      value="a"
    />
    <span
      className="input__underline"
    />
  </div>
</Input>
`;

exports[`Input handleInputBlur updates html correctly 1`] = `
<Input
  onChange={[Function]}
  placeholder="text"
  value=""
>
  <div
    className="input"
  >
    <span
      className="input__header"
    >
      <Text
        center={false}
        color="black"
        size="medium"
        text="text"
        transform="capitalize"
      >
        <div
          className="text text--size-medium text--color-black text--transform-capitalize"
        >
          <span>
            text
          </span>
        </div>
      </Text>
    </span>
    <input
      className="input__body"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="text"
      value=""
    />
    <span
      className="input__underline"
    />
  </div>
</Input>
`;

exports[`Input handleInputFocus updates html correctly 1`] = `
<Input
  onChange={[Function]}
  placeholder="text"
  value=""
>
  <div
    className="input input--focused"
  >
    <span
      className="input__header"
    >
      <Text
        center={false}
        color="black"
        size="medium"
        text="text"
        transform="capitalize"
      >
        <div
          className="text text--size-medium text--color-black text--transform-capitalize"
        >
          <span>
            text
          </span>
        </div>
      </Text>
    </span>
    <input
      className="input__body"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="text"
      value=""
    />
    <span
      className="input__underline"
    />
  </div>
</Input>
`;

exports[`Input render should render correctly 1`] = `
<Input
  onChange={[Function]}
  placeholder="text"
  value=""
>
  <div
    className="input"
  >
    <span
      className="input__header"
    >
      <Text
        center={false}
        color="black"
        size="medium"
        text="text"
        transform="capitalize"
      >
        <div
          className="text text--size-medium text--color-black text--transform-capitalize"
        >
          <span>
            text
          </span>
        </div>
      </Text>
    </span>
    <input
      className="input__body"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="text"
      value=""
    />
    <span
      className="input__underline"
    />
  </div>
</Input>
`;
